plutus-use-cases/test/Spec/vesting.pir

```diff
@@ -22,90 +35,236 @@
       (let
         (nonrec)
         (datatypebind
-          (datatype (tyvardecl Unit (type))  Unit_match (vardecl Unit Unit))
+          (datatype
+            (tyvardecl Bool (type))
```

Comment:
I think this is not an "issue". It happens because the topological sort of the SCCs that algebraic graphs returned us was:
`[SCC(Tuple2) ,SCC(These), SCC(Unit),  SCC(List), SCC(Bool) ,  .... `. The algorithm starts to generate lets from this sequence.  

A kind of trace of the algorithm:
1) generated Tuple2 (nonrec)
2) generated These (nonrec)
3) merged These to Tuple2 (because both are nonrec)
3) generated Unit (nonrec)
5) merged Unit to These-Tuple2 (before)
4) generated List (rec), cannot be merged with before because it is rec, and instead it is generated nested
5) generated Bool (nonrec), cannot be merged with before, because the `List` before is rec. So instead it is nested inside the `List`.

The topological sort is arbitrarily returned by algebraic-graphs. Here there is a better top.sort that yields less groups {Tuple2, These, Unit, Bool, List ...} but we don't know about it.

Reply:
See also Non-guarantees of the algorithm under the Float algorithm note

Reply:
Hm I see. Okay, we can file this as a later improvement. It would be solved by e.g. having multiple binding groups in a let term as we've discussed before, so maybe we can fix it when we do that.